packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,859 @@
+    NETDATA_CLAIM_PATH="${NETDATA_PREFIX}/netdata/usr/sbin/netdata-claim.sh"
+  fi
+
+  if ! pgrep netdata > /dev/null; then
+    NETDATA_CLAIM_EXTRA="${NETDATA_CLAIM_EXTRA} -daemon-not-running"
+  fi
+
+  # shellcheck disable=SC2086
+  if ${ROOTCMD} "${NETDATA_CLAIM_PATH}" -token="${NETDATA_CLAIM_TOKEN}" -rooms="${NETDATA_CLAIM_ROOMS}" -url="${NETDATA_CLAIM_URL}" ${NETDATA_CLAIM_EXTRA}; then
+    progress "Successfully claimed node"
+  else
+    warning "Unable to claim node, you must do so manually."
+    if [ -z "${NETDATA_NEW_INSTALL}" ]; then
+      exit 1
+    fi
+  fi
+}
+
+# ======================================================================
+# Native package install code.
+
+# Check for an already installed package with a given name.
+pkg_installed() {
+  case "${DISTRO_COMPAT_NAME}" in
+    debian|ubuntu)
+      dpkg -l "${1}" > /dev/null 2>&1
+      return $?
+      ;;
+    centos|fedora|opensuse)
+      rpm -q "${1}" > /dev/null 2>&1
+      return $?
+      ;;
+    *)
+      return 1
+      ;;
+  esac
+}
+
+# Check for the existence of a usable package in the repo.
+pkg_avail_check() {
+  case "${DISTRO_COMPAT_NAME}" in
+    debian|ubuntu)
+      env DEBIAN_FRONTEND=noninteractive apt-cache policy netdata | grep -q packagecloud.io/netdata/netdata;
+      return $?
+      ;;
+    centos|fedora)
+      # shellcheck disable=SC2086
+      ${pm_cmd} search -v netdata | grep -qE 'Repo *: netdata(-edge)?$'
+      return $?
+      ;;
+    opensuse)
+      zypper packages -r "$(zypper repos | grep -E 'netdata |netdata-edge ' | cut -f 1 -d '|' | tr -d ' ')" | grep -E 'netdata '
+      return $?
+      ;;
+    *)
+      return 1
+      ;;
+  esac
+}
+
+try_package_install() {
+  if [ -z "${DISTRO}" ]; then
+    warning "Unable to determine Linux distribution for native packages."
+    return 1
+  fi
+
+  progress "Attempting to install using native packages..."
+
+  if [ "${RELEASE_CHANNEL}" = "nightly" ]; then
+    release="-edge"
+  else
+    release=""
+  fi
+
+  if [ "${INTERACTIVE}" = "0" ]; then
+    interactive_opts="-y"
+    env="DEBIAN_FRONTEND=noninteractive"
+  else
+    interactive_opts=""
+    env=""
+  fi
+
+  case "${DISTRO_COMPAT_NAME}" in
+    debian)
+      pm_cmd="apt-get"
+      repo_subcmd="update"
+      repo_prefix="debian/${SYSCODENAME}"
+      pkg_type="deb"
+      pkg_suffix="_all"
+      pkg_vsep="_"
+      pkg_install_opts="${interactive_opts}"
+      repo_update_opts="${interactive_opts}"
+      uninstall_subcmd="uninstall"
+      ;;
+    ubuntu)
+      pm_cmd="apt-get"
+      repo_subcmd="update"
+      repo_prefix="ubuntu/${SYSCODENAME}"
+      pkg_type="deb"
+      pkg_suffix="_all"
+      pkg_vsep="_"
+      pkg_install_opts="${interactive_opts}"
+      repo_update_opts="${interactive_opts}"
+      uninstall_subcmd="uninstall"
+      ;;
+    centos)
+      if command -v dnf > /dev/null; then
+        pm_cmd="dnf"
+        repo_subcmd="makecache"
+      else
+        pm_cmd="yum"
+        repo_subcmd="refresh"
```

Comment:
Hey Austin, it seems in Centos 7.9, `refresh` is not recognized... ? Do we support Cento 7 ?:

```
 --- Updating repository metadata. --- 
[/tmp/netdata-kickstart-lVi2Q95OYZ]$ sudo env yum refresh 
Failed to set locale, defaulting to C
Loaded plugins: fastestmirror
No such command: refresh. Please use /bin/yum --help
 FAILED 
```

Reply:
We _should_ support CentOS 7. I’ll need to look into this further.

Reply:
This should be fixed now.